src/Memoize.jl

```diff
@@ -12,6 +12,12 @@ macro memoize(args...)
         error("Memoize accepts at most two arguments")
     end
 
+    rettype = :nothin
```

Comment:
According to this discourse thread they are equivalent. So I am going to fix this. https://discourse.julialang.org/t/return-type-declarations-what-are-the-semantics-of-f-x-any/4575